hudi-spark-datasource/hudi-spark2/pom.xml

```diff
@@ -197,6 +197,14 @@
       <optional>true</optional>
     </dependency>
 
+    <dependency>
+      <groupId>org.apache.spark</groupId>
+      <artifactId>spark-core_${scala.binary.version}</artifactId>
+      <version>${spark2.version}</version>
+      <scope>provided</scope>
+      <optional>true</optional>
+    </dependency>
```

Comment:
Any reason of adding this?

Reply:
I remember seeing some classpath issues but somehow can't find the exact error message. We can try reverting this change